tests/ci/run_posix_tests.sh

```diff
@@ -40,3 +40,12 @@ if [[ "${AWSLC_FUZZ}" == "1" ]]; then
   echo "Testing building fuzz tests."
   run_build -DFUZZ=1
 fi
+
+# Lightly verify that uncommon build options does not break anything. Fist
+# define a list of typical build options to verify the special build option with
+build_options_to_test=("" "-DBUILD_SHARED_LIBS=1" "-DCMAKE_BUILD_TYPE=Release" "-DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release" "-DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DFIPS=1")
```

Comment:
Move `"-DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DFIPS=1"` to `tests/ci/run_fips_tests.sh`?

The CodeBuild batch spec has FIPS specific dimension, which executes `tests/ci/run_fips_tests.sh`.
https://github.com/awslabs/aws-lc/blob/main/tests/ci/cdk/cdk/codebuild/github_ci_linux_x86_omnibus.yaml#L75-L91

If this change is not made, the build spec can become mixed and the FIPS build is also executed on some env with unsupported(previously not tested) compilers.

Reply:
I would hate having these kind of things spread out over several files... It is already not very easy to figure out what is what, imo.

Reply:
> I would hate having these kind of things spread out over several files... 

Keeps all of these in a single file can increase the build time sequentially. Another blocker is the FIPS build only covers gcc-7 and clang-7 but posix scripts are executed with diff compiler versions.

Regarding the hate or not easy, would you provide suggested changes that can coordinate these?